hack/generator/pkg/astmodel/struct_definition.go

```diff
@@ -10,137 +10,65 @@ import (
 	"go/token"
 )
 
-// StructDefinition encapsulates the definition of a struct
+// StructDefinition encapsulates a complex (object) schema type
 type StructDefinition struct {
-	TypeName         *TypeName
-	StructType       *StructType
-	isResource       bool
-	isStorageVersion bool
-	description      *string
+	typeName    *TypeName
+	structType  *StructType
+	description *string
 }
 
-// IsResource indicates if this is a ARM resource and should be a kubebuilder root
-func (definition *StructDefinition) IsResource() bool {
-	return definition.isResource
+// NewStructDefinition creates a new StructDefinition
+func NewStructDefinition(typeName *TypeName, structType *StructType) *StructDefinition {
+	return &StructDefinition{typeName, structType, nil}
 }
 
-// Ensure StructDefinition implements TypeDefiner interface correctly
-var _ TypeDefiner = (*StructDefinition)(nil)
+// ensure StructDefinition is a TypeDefiner
+var _ TypeDefiner = &StructDefinition{}
```

Comment:
Why are we changing our approach here? I thought the `(*Type)(nil)` approach was idiomatic for Go.

Reply:
it’s harder to type!